src/SemiclassicalOrthogonalPolynomials.jl

```diff
@@ -260,56 +268,110 @@ function semijacobi_ldiv(P::SemiclassicalJacobi, Q)
+Numerically unstable if the parameter modification is large. Typically one should instead use `P \\ Q` which is equivalent to `semijacobi_ldiv(P,Q)` and proceeds step by step.
+"""
+function semijacobi_ldiv_direct(Q::SemiclassicalJacobi, P::SemiclassicalJacobi)
+    (Q.t ≈ P.t) && (Q.a ≈ P.a) && (Q.b ≈ P.b) && (Q.c ≈ P.c) && return SymTridiagonal(Ones(∞),Zeros(∞))
+    Δa = Q.a-P.a
+    Δb = Q.b-P.b
+    Δc = Q.c-P.c
+    # special case (Δa,Δb,Δc) = (2,0,0)
+    if (Δa == 2) && iszero(Δb) && iszero(Δc)
+        M = qr(P.X).R
+        return ApplyArray(*, Diagonal(sign.(view(M,band(0)))/abs(M[1])), M)
+    # special case (Δa,Δb,Δc) = (0,2,0)
+    elseif iszero(Δa) && (Δb == 2) && iszero(Δc)
+        M = qr(I-P.X).R
+        return ApplyArray(*, Diagonal(sign.(view(M,band(0)))/abs(M[1])), M)
+    # special case (Δa,Δb,Δc) = (0,0,2)
+    elseif iszero(Δa) && iszero(Δb) && (Δc == 2)
+        M = qr(Q.t*I-P.X).R
+        return ApplyArray(*, Diagonal(sign.(view(M,band(0)))/abs(M[1])), M)
+    # special case (Δa,Δb,Δc) = (-2,0,0)
+    elseif (Δa == -2) && iszero(Δb) && iszero(Δc)
+        M = qr(Q.X).R
+        return ApplyArray(\, M, Diagonal(sign.(view(M,band(0)))*abs.(M[1])))
+    # special case (Δa,Δb,Δc) = (0,-2,0)
+    elseif iszero(Δa) && (Δb == -2) && iszero(Δc)
+        M = qr(I-Q.X).R
+        return ApplyArray(\, M, Diagonal(sign.(view(M,band(0)))*abs.(M[1])))
+    # special case (Δa,Δb,Δc) = (0,0,-2)
+    elseif iszero(Δa) && iszero(Δb) && (Δc == -2)
+        M = qr(Q.t*I-Q.X).R
+        return ApplyArray(\, M, Diagonal(sign.(view(M,band(0)))*abs.(M[1])))
+    # special case (Δa,Δb,Δc) = (1,0,0)
+    elseif isone(Δa) && iszero(Δb) && iszero(Δc)
         M = cholesky(P.X).U
         return ApplyArray(*, Diagonal(Fill(1/M[1],∞)), M)
-    elseif iszero(Δa) && isone(Δb) && iszero(Δc) # special case (Δa,Δb,Δc) = (0,1,0)
+    # special case (Δa,Δb,Δc) = (0,1,0)
+    elseif iszero(Δa) && isone(Δb) && iszero(Δc)
         M = cholesky(I-P.X).U
         return ApplyArray(*, Diagonal(Fill(1/M[1],∞)), M)
-    elseif iszero(Δa) && iszero(Δb) && isone(Δc) # special case (Δa,Δb,Δc) = (0,0,1)
-        M = cholesky(Qt*I-P.X).U
+    # special case (Δa,Δb,Δc) = (0,0,1)
+    elseif iszero(Δa) && iszero(Δb) && isone(Δc)
+        M = cholesky(Q.t*I-P.X).U
+        return ApplyArray(*, Diagonal(Fill(1/M[1],∞)), M)
+    # special case (Δa,Δb,Δc) = (-1,0,0)
+    elseif isone(-Δa) && iszero(Δb) && iszero(Δc)
+        M = cholesky(Q.X).U
+        return ApplyArray(\, M, Diagonal(Fill(M[1],∞)))
+    # special case (Δa,Δb,Δc) = (0,-1,0)
+    elseif iszero(Δa) && isone(-Δb) && iszero(Δc)
+        M = cholesky(I-Q.X).U
+        return ApplyArray(\, M, Diagonal(Fill(M[1],∞)))
+    # special case (Δa,Δb,Δc) = (0,0,-1)
+    elseif iszero(Δa) && iszero(Δb) && isone(-Δc)
+        M = cholesky(Q.t*I-Q.X).U
+        return ApplyArray(\, M, Diagonal(Fill(M[1],∞)))
```

Comment:
```suggestion
        return inv(M/M[1])
```

Reply:
`inv` doesn't work when applied to this type. `ApplyArray(inv,M/M[1])` works similarly to `ApplyArray(\, M, Diagonal(Fill(M[1],∞)))` though Julia doesn't seem to know it's Triangular so it causes some odd behavior here and there. This feels like something to fix elsewhere and then we can make it just read `inv(M/M[1])` here?

For the record, with  `ApplyArray(inv,M/M[1])`  we get
```julia
MemoryLayout(R_0)
LazyArrays.InvLayout{TriangularLayout{'U', 'N', LazyArrays.LazyLayout}}()
```

with `inv(M/M[1])` it errors:
```julia
julia>         R_0 = SemiclassicalJacobi(t, 1, 2, 2) \ P
ERROR: MethodError: no method matching Matrix{Float64}(::UniformScaling{Bool}, ::Tuple{Infinities.InfiniteCardinal{0}, Infinities.InfiniteCardinal{0}})

Closest candidates are:
  Matrix{T}(::UndefInitializer, ::Tuple{Union{Infinities.InfiniteCardinal{0}, Infinities.Infinity}, Union{Infinities.InfiniteCardinal{0}, Infinities.Infinity}}) where T
   @ InfiniteArrays C:\Users\timon\.julia\packages\InfiniteArrays\heAfT\src\infarrays.jl:7
  Matrix{T}(::UndefInitializer, ::Tuple{Integer, Union{Infinities.InfiniteCardinal{0}, Infinities.Infinity}}) where T
   @ InfiniteArrays C:\Users\timon\.julia\packages\InfiniteArrays\heAfT\src\infarrays.jl:4
  Matrix{T}(::UndefInitializer, ::Tuple{Union{Infinities.InfiniteCardinal{0}, Infinities.Infinity}, Integer}) where T
   @ InfiniteArrays C:\Users\timon\.julia\packages\InfiniteArrays\heAfT\src\infarrays.jl:5
  ...

Stacktrace:
 [1] inv(A::UpperTriangular{Float64, BroadcastMatrix{Float64, typeof(/), Tuple{…}}})
   @ LinearAlgebra C:\Users\timon\AppData\Local\Programs\Julia-1.10.0\share\julia\stdlib\v1.10\LinearAlgebra\src\triangular.jl:797    
 [2] semijacobi_ldiv_direct(Q::SemiclassicalJacobi{Float64}, P::SemiclassicalJacobi{Float64})
   @ SemiclassicalOrthogonalPolynomials c:\Users\timon\OneDrive\Documents\Work projects\Archive\SemiclassicalOrthogonalPolynomials.jl\src\SemiclassicalOrthogonalPolynomials.jl:322
 [3] semijacobi_ldiv(Q::SemiclassicalJacobi{Float64}, P::SemiclassicalJacobi{Float64})
   @ SemiclassicalOrthogonalPolynomials c:\Users\timon\OneDrive\Documents\Work projects\Archive\SemiclassicalOrthogonalPolynomials.jl\src\SemiclassicalOrthogonalPolynomials.jl:352
 [4] \(A::SemiclassicalJacobi{Float64}, B::SemiclassicalJacobi{Float64})
   @ SemiclassicalOrthogonalPolynomials c:\Users\timon\OneDrive\Documents\Work projects\Archive\SemiclassicalOrthogonalPolynomials.jl\src\SemiclassicalOrthogonalPolynomials.jl:411
 [5] top-level scope
   @ c:\Users\timon\OneDrive\Documents\Work projects\Archive\SemiclassicalOrthogonalPolynomials.jl\test\runtests.jl:663
Some type information was truncated. Use `show(err)` to see complete types.
```